dbc2val/requirements.in

```diff
@@ -0,0 +1,25 @@
+# This file contains top level dependencies pinning a major version.
+# For developing the dbc2val package it makes sense to use this file
+# with 'pip install -r requirements.in' in order to benefit from more
```

Comment:
We need to agree on how we want to handle dev-dependencies. So far requirements-dev.txt and requirements.txt for dbc2val has been quite similar. But as part of #52 I will need to add pylint and pytest, but they should only be used for building/testing, they are not required for running and does not for instance need to be installed in the docker container.

Shall we then (as part of #52) re-introduce a file `requirements-dev.in` or similar that only contains the extra dependencies for build/test, i.e. when developing you should do `pip install -r requirements.in requirements-dev.in` or similar?

Reply:
I am not expert in Python build systems but something a little more sophisticated than plain `pip install` would probably be nice. [Hatch](https://hatch.pypa.io/latest/) seems to be quite popular for these purposes and also seems to provide mechanisms for creating (virtual) environments easily based on metadata define in a standard *pyproject.toml* file. It also provides means for specifying *additional* dependencies required only for testing etc. Maybe it would be worth a shot?

Reply:
In kuksa-client the last thing we tried was pip-compile from https://pypi.org/project/pip-tools/  or https://github.com/jazzband/pip-tools

We don't have much experience yet, but it was chosen as middleground between "naked" requirements and full pipenv Pipfile/Pipfile.lock combo.

So the requirments are generated and checked in as in here https://github.com/eclipse/kuksa.val/blob/master/kuksa-client/requirements.txt.

In that case the source was setup.cfg, but a requirements.in is also possible (although I think then you need to have requirements.in and  requirements-dev.in separately, with a toml might be better) 

No hard opinion here, just one datapoint :) My only opinion is, a requirements file without ANY hint as to what version we would like is not a best-practice 😃